### ISSUE TITLE
PairWise Ranking Loss Implementation

### ISSUE THREAD

**Issue by gautamMalu** [aimerykong/deepImageAestheticsAnalysis#1]
In the paper, you have mentioned custom ranking loss 
lossrank = 1/2N Σmax(0,α−δ(yi ≥ yj)(ˆyi − ˆyj))

The standard Caffe toolbox doesn't support this loss function. I have looked around and I have found triplet loss function's implementation https://github.com/wanji/caffe-sl/ but not this custom pairwise ranking loss.  
Would you be kind enough to share your implementation of this loss function?

**Comment by aimerykong** [comment]
Hi, Gautam,

Thanks for your interest in our work.

I modified siamese function in caffe for the rank loss layer. This part was
done in Adobe. So there are copyright issues.

Please let me re-write it and upload later today and I'll let you know soon.

Regards,
Shu

**Comment by aimerykong** [comment]
hi, Gautam,

The modified caffe is zipped (caffeCustom.zip) and can be found thru

https://drive.google.com/open?id=0BxeylfSgpk1MOVduWGxyVlJFUHM


An example of training prototxt can also be found there, named
"mergedNetRank.prototxt"


Please let me know there is any problem. Hope this helps.


Regards,

Shu

**Comment by gautamMalu** [comment]
Thanks, I will check it out.
